### DISCUSSION TITLE
Jetbrains Rider XAML "Errors"

### DISCUSSION THREAD

**Discussion by lylerolleman** [unoplatform/uno#13335] [General]
Following the guide at https://platform.uno/docs/articles/get-started-rider.html?tabs=windows I am seeing design time errors in the XAML files created in the template

Ex. `<utu:NavigationBar Content="{Binding Title}" />` is not recognized and wants to change it to use `xmlns:ui="clr-namespace:Uno.Toolkit.UI;assembly=Uno.Toolkit.WinUI"`, which results in a build error. 

VS does not seem to have the same issue, and is only a problem at design time, it builds just fine. 

I'm running Rider 2023.1.2

**Comment by jeromelaban** [comment]
Thanks for the report. At this time, Rider does not understand the WinUI XAML properly and will report design-time errors. You can comment on this issue to make your voice heard: https://youtrack.jetbrains.com/issue/RIDER-69976/Please-Improve-Support-for-Uno-Platform

To edit XAML with code completion, Visual Studio and Visual Studio code are the two alternatives at this time.

**Reply by mrc-eu** [reply]
What's the current state of Uno support in Rider? Is it usable; that is: supporting Hot Reload and XAML?

Or is it better to VSCode for now, which I try to avoid.

**Reply by jeromelaban** [reply]
Rider support has improved, XAML code completion is now working properly. Hot Reload and [WebAssembly debugging](https://youtrack.jetbrains.com/issue/RIDER-103346/Uno-Platform-for-WebAssembly-debugger-support) is still not supported at this time (make sure to upvote the webassembly debugging issue!).
